src/SFML/Graphics/Shader.cpp

```diff
@@ -882,9 +883,9 @@ bool Shader::compile(std::string_view vertexShaderCode, std::string_view geometr
         glCheck(GLEXT_glGetObjectParameteriv(vertexShader, GLEXT_GL_OBJECT_COMPILE_STATUS, &success));
         if (success == GL_FALSE)
         {
-            char log[1024];
-            glCheck(GLEXT_glGetInfoLog(vertexShader, sizeof(log), nullptr, log));
-            err() << "Failed to compile vertex shader:" << '\n' << log << std::endl;
+            std::array<char, 1024> log{};
+            glCheck(GLEXT_glGetInfoLog(vertexShader, sizeof(log), nullptr, log.data()));
```

Comment:
`log.size()` would be more obviously correct than `sizeof(log)`

Reply:
`std::array:size` says "I want the number of elements". `sizeof` says "I want the number of bytes". In the case of `std::array<char>` and similar specializations, the two both return the same value but I still think they convey different meanings. In the case of C APIs that want a size (in bytes) and an address, I think it makes sense to use `sizeof`, but we can still use `.size()` without issue since realistically nobody will refine `log` to be an array of `wchar_t` or something else larger than 1 byte.

Reply:
My thinking was that `sizeof(some_container)` is generally not related to the valid range `[data(), data() + size())`. It happens to be ok with `std::array<char>` but it is dependent on the kind of container. It would not work with `std::vector<char>`. Anyways this is minor, let's ignore this if I am the only one confused with this `sizeof`.

Reply:
> It happens to be ok with std::array<char> but it is dependent on the kind of container. It would not work with std::vector<char>.

For whatever reason that hadn't crossed my mind. I was guarding against the admittedly less likely situation where the array type was changed to something larger than 1 byte. I see no reason that would ever happen but I can certainly imagine a situation where the container is replaced with `std::vector`. I think I'm coming around to your perspective on this.